### ISSUE TITLE
Restreaming to Facebook is not working with v2.8.0

### ISSUE THREAD

**Issue by yashtandon113** [ant-media/Ant-Media-Server#5961]
### Short description
Restreaming to Facebook is not working with v2.8.0 while it is working fine with YouTube. In v2.7.0, restreaming to FB is also working fine.

### Environment
* Ant Media Server version: v2.8.0 snapshot

### Steps to reproduce
1. Create a live stream on a server
2. Add the FB rtmp endpoint
3. Publish the stream to the created stream ID.

### Expected behavior
Stream should be republished on Facebook.

### Actual behavior
Stream is not republishing on Facebook.
 
```
2023-12-22 19:59:19,128 [vert.x-worker-thread-18] INFO  i.a.AntMediaApplicationAdapter -  Status of stream test is set to Broadcasting with result: true
2023-12-22 19:59:19,768 [vert.x-worker-thread-18] WARN  io.antmedia.muxer.RtmpMuxer - Not writing AudioBuffer for test because Is running:false
2023-12-22 19:59:20,328 [vert.x-eventloop-thread-4] INFO  i.a.AntMediaApplicationAdapter - Broadcast absolute time is not available for stream:test
2023-12-22 19:59:21,126 [vert.x-eventloop-thread-4] INFO  io.antmedia.muxer.MuxAdaptor - Checking the endpoint health for: rtmps://live-api-s.facebook.com:443/rtmp/FB-6777937405638300-0-AbwP9D0WdQEEgsNT and status: failed 
2023-12-22 19:59:21,126 [vert.x-eventloop-thread-4] INFO  io.antmedia.muxer.MuxAdaptor - Endpoint check returned error for 1 times for endpoint rtmps://live-api-s.facebook.com:443/rtmp/FB-6777937405638300-0-AbwP9D0WdQEEgsNT
2023-12-22 19:59:21,239 [vert.x-worker-thread-96] WARN  io.antmedia.muxer.RtmpMuxer - Not writing AudioBuffer for test because Is running:false
2023-12-22 19:59:22,718 [vert.x-worker-thread-15] WARN  io.antmedia.muxer.RtmpMuxer - Not writing to RTMP muxer because it's not started for rtmps://live-api-s.facebook.com:443/rtmp/FB-6777937405638300-0-AbwP9D0WdQEEgsNT
2023-12-22 19:59:23,126 [vert.x-eventloop-thread-4] INFO  io.antmedia.muxer.MuxAdaptor - Checking the endpoint health for: rtmps://live-api-s.facebook.com:443/rtmp/FB-6777937405638300-0-AbwP9D0WdQEEgsNT and status: failed 
2023-12-22 19:59:23,127 [vert.x-eventloop-thread-4] INFO  io.antmedia.muxer.MuxAdaptor - Endpoint check returned error for 2 times for endpoint rtmps://live-api-s.facebook.com:443/rtmp/FB-6777937405638300-0-AbwP9D0WdQEEgsNT
2023-12-22 19:59:23,339 [vert.x-worker-thread-6] INFO  io.antmedia.muxer.MuxAdaptor - Stream queue size:0 speed:1.0038803926044282 for streamId:test 
2023-12-22 19:59:24,179 [vert.x-worker-thread-48] WARN  io.antmedia.muxer.RtmpMuxer - Not writing AudioBuffer for test because Is running:false
2023-12-22 19:59:25,126 [vert.x-eventloop-thread-4] INFO  io.antmedia.muxer.MuxAdaptor - Checking the endpoint health for: rtmps://live-api-s.facebook.com:443/rtmp/FB-6777937405638300-0-AbwP9D0WdQEEgsNT and status: failed 
2023-12-22 19:59:25,127 [vert.x-eventloop-thread-4] INFO  io.antmedia.muxer.MuxAdaptor - Health check process failed, trying to republish to the endpoint: rtmps://live-api-s.facebook.com:443/rtmp/FB-6777937405638300-0-AbwP9D0WdQEEgsNT
2023-12-22 19:59:25,128 [vert.x-eventloop-thread-4] INFO  io.antmedia.muxer.RtmpMuxer - Not writing trailer because header is not written yet
2023-12-22 19:59:25,128 [vert.x-eventloop-thread-4] INFO  io.antmedia.muxer.MuxAdaptor - Endpoint status updated to finished  for streamId: test for url: rtmps://live-api-s.facebook.com:443/rtmp/FB-6777937405638300-0-AbwP9D0WdQEEgsNT
2023-12-22 19:59:25,129 [vert.x-eventloop-thread-4] INFO  io.antmedia.muxer.MuxAdaptor - start rtmp streaming for stream id:test to rtmps://live-api-s.facebook.com:443/rtmp/FB-6777937405638300-0-AbwP9D0WdQEEgsNT with requested resolution height720 stream resolution:720
2023-12-22 19:59:25,132 [vert.x-eventloop-thread-4] INFO  io.antmedia.muxer.MuxAdaptor - prepareMuxer for stream:test muxer:RtmpMuxer
2023-12-22 19:59:25,133 [vert.x-eventloop-thread-4] INFO  io.antmedia.muxer.MuxAdaptor - Add video stream to muxer:RtmpMuxer for streamId:test
2023-12-22 19:59:25,134 [vert.x-eventloop-thread-4] INFO  io.antmedia.muxer.RtmpMuxer - Creating outputFormatContext
2023-12-22 19:59:25,135 [vert.x-eventloop-thread-4] INFO  io.antmedia.muxer.RtmpMuxer - Adding timebase to the input time base map index:0 value: 1/1000 for stream:test type:video
2023-12-22 19:59:25,136 [vert.x-eventloop-thread-4] INFO  io.antmedia.muxer.MuxAdaptor - Add audio stream to muxer:RtmpMuxer for streamId:test
2023-12-22 19:59:25,137 [vert.x-eventloop-thread-4] INFO  io.antmedia.muxer.RtmpMuxer - Adding timebase to the input time base map index:1 value: 1/1000 for stream:test type:audio
2023-12-22 19:59:25,139 [vert.x-eventloop-thread-4] INFO  io.antmedia.muxer.MuxAdaptor - Muxer:RtmpMuxer is prepared succesfully for streamId:test
2023-12-22 19:59:25,141 [vert.x-worker-thread-109] WARN  io.antmedia.muxer.RtmpMuxer - Not writing AudioBuffer for test because Is running:false
2023-12-22 19:59:26,599 [vert.x-worker-thread-30] WARN  io.antmedia.muxer.RtmpMuxer - Not writing AudioBuffer for test because Is running:false
2023-12-22 19:59:26,617 [vert.x-worker-thread-117] WARN  io.antmedia.muxer.RtmpMuxer - Could not open output url: rtmps://live-api-s.facebook.com:443/rtmp/FB-6777937405638300-0-AbwP9D0WdQEEgsNT 
2023-12-22 19:59:26,618 [vert.x-worker-thread-117] INFO  io.antmedia.muxer.MuxAdaptor - Endpoint status updated to failed  for streamId: test for url: rtmps://live-api-s.facebook.com:443/rtmp/FB-6777937405638300-0-AbwP9D0WdQEEgsNT
2023-12-22 19:59:26,619 [vert.x-worker-thread-117] ERROR io.antmedia.muxer.RtmpMuxer - Cannot initializeOutputFormatContextIO for rtmp endpoint:rtmps://live-api-s.facebook.com:443/rtmp/FB-6777937405638300-0-AbwP9D0WdQEEgsNT
2023-12-22 19:59:28,078 [vert.x-worker-thread-61] WARN  io.antmedia.muxer.RtmpMuxer - Not writing to RTMP muxer because it's not started for rtmps://live-api-s.facebook.com:443/rtmp/FB-6777937405638300-0-AbwP9D0WdQEEgsNT
2023-12-22 19:59:28,358 [vert.x-worker-thread-7] INFO  io.antmedia.muxer.MuxAdaptor - Stream queue size:1 speed:0.9995454545454545 for streamId:test 
2023-12-22 19:59:28,620 [vert.x-eventloop-thread-4] INFO  io.antmedia.muxer.MuxAdaptor - Checking the endpoint health for: rtmps://live-api-s.facebook.com:443/rtmp/FB-6777937405638300-0-AbwP9D0WdQEEgsNT and status: failed 
2023-12-22 19:59:28,621 [vert.x-eventloop-thread-4] INFO  io.antmedia.muxer.MuxAdaptor - Endpoint check returned error for 1 times for endpoint rtmps://live-api-s.facebook.com:443/rtmp/FB-6777937405638300-0-AbwP9D0WdQEEgsNT
2023-12-22 19:59:29,539 [vert.x-worker-thread-99] WARN  io.antmedia.muxer.RtmpMuxer - Not writing AudioBuffer for test because Is running:false
2023-12-22 19:59:30,620 [vert.x-eventloop-thread-4] INFO  io.antmedia.muxer.MuxAdaptor - Checking the endpoint health for: rtmps://live-api-s.facebook.com:443/rtmp/FB-6777937405638300-0-AbwP9D0WdQEEgsNT and status: failed 
2023-12-22 19:59:30,621 [vert.x-eventloop-thread-4] INFO  io.antmedia.muxer.MuxAdaptor - Endpoint check returned error for 2 times for endpoint rtmps://live-api-s.facebook.com:443/rtmp/FB-6777937405638300-0-AbwP9D0WdQEEgsNT
2023-12-22 19:59:31,012 [vert.x-worker-thread-9] WARN  io.antmedia.muxer.RtmpMuxer - Not writing AudioBuffer for test because Is running:false
2023-12-22 19:59:32,489 [vert.x-worker-thread-57] WARN  io.antmedia.muxer.RtmpMuxer - Not writing AudioBuffer for test because Is running:false
2023-12-22 19:59:32,619 [vert.x-eventloop-thread-4] INFO  io.antmedia.muxer.MuxAdaptor - Checking the endpoint health for: rtmps://live-api-s.facebook.com:443/rtmp/FB-6777937405638300-0-AbwP9D0WdQEEgsNT and status: failed 
2023-12-22 19:59:32,620 [vert.x-eventloop-thread-4] INFO  io.antmedia.muxer.MuxAdaptor - Health check process failed, trying to republish to the endpoint: rtmps://live-api-s.facebook.com:443/rtmp/FB-6777937405638300-0-AbwP9D0WdQEEgsNT
2023-12-22 19:59:32,620 [vert.x-eventloop-thread-4] INFO  io.antmedia.muxer.RtmpMuxer - Not writing trailer because header is not written yet
2023-12-22 19:59:32,621 [vert.x-eventloop-thread-4] INFO  io.antmedia.muxer.MuxAdaptor - Endpoint status updated to finished  for streamId: test for url: rtmps://live-api-s.facebook.com:443/rtmp/FB-6777937405638300-0-AbwP9D0WdQEEgsNT
2023-12-22 19:59:32,622 [vert.x-eventloop-thread-4] INFO  io.antmedia.muxer.MuxAdaptor - start rtmp streaming for stream id:test to rtmps://live-api-s.facebook.com:443/rtmp/FB-6777937405638300-0-AbwP9D0WdQEEgsNT with requested resolution height720 stream resolution:720
2023-12-22 19:59:32,623 [vert.x-eventloop-thread-4] INFO  io.antmedia.muxer.MuxAdaptor - prepareMuxer for stream:test muxer:RtmpMuxer
2023-12-22 19:59:32,624 [vert.x-eventloop-thread-4] INFO  io.antmedia.muxer.MuxAdaptor - Add video stream to muxer:RtmpMuxer for streamId:test
2023-12-22 19:59:32,625 [vert.x-eventloop-thread-4] INFO  io.antmedia.muxer.RtmpMuxer - Creating outputFormatContext
2023-12-22 19:59:32,626 [vert.x-eventloop-thread-4] INFO  io.antmedia.muxer.RtmpMuxer - Adding timebase to the input time base map index:0 value: 1/1000 for stream:test type:video
2023-12-22 19:59:32,626 [vert.x-eventloop-thread-4] INFO  io.antmedia.muxer.MuxAdaptor - Add audio stream to muxer:RtmpMuxer for streamId:test
2023-12-22 19:59:32,627 [vert.x-eventloop-thread-4] INFO  io.antmedia.muxer.RtmpMuxer - Adding timebase to the input time base map index:1 value: 1/1000 for stream:test type:audio
2023-12-22 19:59:32,628 [vert.x-eventloop-thread-4] INFO  io.antmedia.muxer.MuxAdaptor - Muxer:RtmpMuxer is prepared succesfully for streamId:test
2023-12-22 19:59:32,645 [vert.x-worker-thread-91] WARN  io.antmedia.muxer.RtmpMuxer - Not writing to RTMP muxer because it's not started for rtmps://live-api-s.facebook.com:443/rtmp/FB-6777937405638300-0-AbwP9D0WdQEEgsNT
2023-12-22 19:59:32,989 [vert.x-worker-thread-41] INFO  i.a.e.adaptive.StreamAdaptor - Stream adaptor queue size: 0 for stream: test
2023-12-22 19:59:33,363 [vert.x-worker-thread-95] INFO  io.antmedia.muxer.MuxAdaptor - Stream queue size:0 speed:0.9990917347865577 for streamId:test 
2023-12-22 19:59:33,489 [vert.x-worker-thread-81] WARN  io.antmedia.muxer.RtmpMuxer - Could not open output url: rtmps://live-api-s.facebook.com:443/rtmp/FB-6777937405638300-0-AbwP9D0WdQEEgsNT 
2023-12-22 19:59:33,490 [vert.x-worker-thread-81] INFO  io.antmedia.muxer.MuxAdaptor - Endpoint status updated to failed  for streamId: test for url: rtmps://live-api-s.facebook.com:443/rtmp/FB-6777937405638300-0-AbwP9D0WdQEEgsNT
2023-12-22 19:59:33,490 [vert.x-worker-thread-81] ERROR io.antmedia.muxer.RtmpMuxer - Cannot initializeOutputFormatContextIO for rtmp endpoint:rtmps://live-api-s.facebook.com:443/rtmp/FB-6777937405638300-0-AbwP9D0WdQEEgsNT
```

**Comment by nocpbxchap** [comment]
any update about these, because not working also on 2.8.2, regards.

**Comment by yashtandon113** [comment]
@nocpbxchap 
Yes, you are right. Thanks for reporting the issue again.

Some changes were made to resolve the issue, but it has reoccurred. We will investigate and reschedule the issue in order to resolve it.

**Comment by yogeshJain0727** [comment]
@yashtandon113 We are facing the same issue with Facebook and LinkedIn. Any updates regarding this?

**Comment by brixxnatt** [comment]
any updates ?

**Comment by brixxnatt** [comment]
UP

**Comment by mekya** [comment]
Hi @brixxnatt and @yashtandon113,

PR has been merged for this fix. Please try community edition on sonatype to test.
https://oss.sonatype.org/#nexus-search;gav~io.antmedia~ant-media-server~~~~kw,versionexpand 

Let me know if it works for you

Cheers

**Comment by haiktqd** [comment]
I have same problem (2.8.2). So, which version is the best?

**Comment by nocpbxchap** [comment]
Version 2.8.0 only works for kick for others i am using old 2.5.3 that
works for all but not for kick i think when fixing that issue didnt test
the others . Check others.

**Comment by yashtandon113** [comment]
@haiktqd @nocpbxchap 
The Facebook re-streaming issue has been resolved in the latest snapshot, version 2.8.3 of AMS. The ETA for the 2.8.3 release is early April.

If you are using the Community Edition of Ant Media Server, then you can either download the latest snapshot from [here](https://oss.sonatype.org/#nexus-search;gav~io.antmedia~ant-media-server~~~~kw,versionexpand) or you can wait for the 2.8.3 official release.

For Enterprise Edition, you can download the latest snapshot from your [antmedia.io](https://antmedia.io) account.

To upgrade your server, check [this document](https://antmedia.io/docs/guides/installing-on-linux/upgrading-ant-media-server/).
